Packs/AzureCompute/Integrations/AzureCompute_v2/AzureCompute_v2.py

```diff
@@ -382,6 +407,11 @@ def poweroff_vm(self, resource_group, vm_name):
         return self.ms_client.http_request(
             method='POST', url_suffix=url_suffix, params=parameters, resp_type="response")
 
+    def get_all_public_ip_details(self):
+        url_suffix = "/providers/Microsoft.Network/publicIPAddresses"
+        parameters = {'api-version': API_VERSION}
+        return self.ms_client_no_resource_group.http_request(method='GET', url_suffix=url_suffix, params=parameters)
```

Comment:
@omerKarkKatz , i don't understand the reasoning for the change here.  Is it to not use the `ms_client_no_resource_group` object?  This was added so that we don't need to have the resource group to look up public IP information.

Reply:
@johnnywilkes if you will look carefully the only difference between the 'ms_client_no_resource_group' and the 'ms_client' configured in the MSGraphClient is the base-url parameter. I think it is redundant to configure a new instance. I just overridden the base_url of the ms.client with the base URL you defined in the  'ms_client_no_resource_group' and initialized the ms_client. I think it should have the exact same behavior. Please correct me if wrong.

Reply:
@omerKarkKatz , okay, that makes sense.  Would you recommendation me in main function to do some logic such as?:
```
if command == 'azure-vm-get-all-public-ip-details':
    base_url = f"{server}/subscriptions/{subscription_id}/"
else:
    base_url = f"{server}/subscriptions/{subscription_id}/resourceGroups/"
```

Reply:
@omerKarkKatz good suggestion, I wanted something simpler like this. For this change did you mean to use `_base_url`, like `self.ms_client._base_url`, since base_url is an internal variable separate from the MicrosoftClient object?

Reply:
@BigEasyJ Sorry i didn't really understand what you were meaning. we can talk about this in the Demo.